src/transformers/models/musicgen/modeling_musicgen.py

```diff
@@ -2170,6 +2170,24 @@ def _maybe_initialize_input_ids_for_generation(
                 break
         return torch.ones((batch_size, 1), dtype=torch.long, device=self.device) * bos_token_id
 
+    def _get_decoder_start_token_id(
```

Comment:
Musicgen (and its melody variant) have their own custom `generate`, relying on this method. 

I've intentionally not updated this custom `generate`, to pressure us into moving towards a single generate function.